internal/gw/gw.go

```diff
@@ -58,7 +58,7 @@ type Client interface {
 	// This function never fails, but it is not guaranteed that the publish object
```

Comment:
This info in the comment is now incorrect, since the design has changed from "it never fails" to "it can fail".  Could it please be rewritten so it's accurate?